src/sdltiles.cpp

```diff
@@ -4123,3 +4062,64 @@ HWND getWindowHandle()
 #endif
 
 #endif // TILES
+
+static window_dimensions get_window_dimensions( const catacurses::window &win,
+        const point &pos, const point &size )
+{
+    window_dimensions dim;
+    if( use_tiles && g && win == g->w_terrain ) {
+        // tiles might have different dimensions than standard font
+        dim.scaled_font_size.x = tilecontext->get_tile_width();
+        dim.scaled_font_size.y = tilecontext->get_tile_height();
+    } else if( map_font && g && win == g->w_terrain ) {
+        // map font (if any) might differ from standard font
+        dim.scaled_font_size.x = map_font->fontwidth;
+        dim.scaled_font_size.y = map_font->fontheight;
+    } else if( overmap_font && g && win == g->w_overmap ) {
+        dim.scaled_font_size.x = overmap_font->fontwidth;
+        dim.scaled_font_size.y = overmap_font->fontheight;
+    } else {
+        dim.scaled_font_size.x = fontwidth;
+        dim.scaled_font_size.y = fontheight;
+    }
+#if defined (TILES)
+    // multiplied by the user's specified scaling factor regardless of whether tiles are in use
+    dim.scaled_font_size *= get_scaling_factor();
+#endif
+    if( win ) {
+        cata_cursesport::WINDOW *const pwin = win.get<cata_cursesport::WINDOW>();
+        dim.window_pos_cell = pwin->pos;
+        dim.window_size_cell.x = pwin->width;
+        dim.window_size_cell.y = pwin->height;
+    } else {
+        dim.window_pos_cell = pos;
+        dim.window_size_cell = size;
+    }
+
+    // the window position is *always* in standard font dimensions!
+    dim.window_pos_pixel = point( dim.window_pos_cell.x * fontwidth,
+                                  dim.window_pos_cell.y * fontheight );
+    // But the size of the window is in the font dimensions of the window.
+    dim.window_size_pixel.x = dim.window_size_cell.x * dim.scaled_font_size.x;
+    dim.window_size_pixel.y = dim.window_size_cell.y * dim.scaled_font_size.y;
+
+    return dim;
+}
+
+window_dimensions get_window_dimensions( const catacurses::window &win )
+{
+    return get_window_dimensions( win, point_zero, point_zero );
+}
+
+window_dimensions get_window_dimensions( const point &pos, const point &size )
+{
+    return get_window_dimensions( {}, pos, size );
+}
+
+bool window_contains_point_relative( const catacurses::window &win, const point &p )
+{
+    const window_dimensions dim = get_window_dimensions( win );
```

Comment:
You can instead use `getmaxx` and `getmaxy` to get the text size of the window here.

Reply:
> You can instead use `getmaxx` and `getmaxy` to get the text size of the window here.

is it OK now

Reply:
You can just use the code inside `#if defined(CURSES)` for both SDL and curses.

Reply:
> You can just use the code inside `#if defined(CURSES)` for both SDL and curses.

so what is the point of having get_window_dimensions while we already have functions under catacurses such as getmaxx and getmaxy?

Reply:
Because `get_window_dimensions` not only returns the size in characters, but also in pixels, but on curses there's not such thing as a pixel.